src/generators/randgraphs.jl

```diff
@@ -598,6 +598,28 @@ function random_regular_digraph(n::Integer, k::Integer; dir::Symbol=:out, seed::
     end
 end
 
+@doc_str """
+    random_tournament_digraph(n)
+
+Create a random directed [tournament graph]
+(https://en.wikipedia.org/wiki/Tournament_(graph_theory))
+with `n` vertices.
+
+### Optional Arguments
+- `seed=-1`: set the RNG seed.
+"""
+function random_tournament_digraph(n::Integer; seed::Int=-1)
+
+    rng = getRNG(seed)
+    g = DiGraph(n)
+
+    for i = 1:n, j = i+1:n
+        rand(rng, Bool) ? add_edge!(g, Edge(i, j)) : add_edge!(g, Edge(j, i))
```

Comment:
I like more
```
add_edge!(g, rand(rng, Bool) ?  Edge(i, j) : Edge(j, i))
```

Reply:
I don't like that syntax as much, since I have to go into the `add_edge` to understand there's a conditional. Is there a performance reason to do this, or is it just syntactical preference?

Reply:
I do not think so, but possibly `add_edge!(g, ifelse(rand(rng, Bool), Edge(i, j), Edge(j, i)))` can avoid a branch. "In some cases, using `ifelse` instead of an if statement can eliminate
  the branch in generated code and provide higher performance in tight loops." A matter of trying.

Reply:
I'm not sure you can avoid branches here given that we're determining a branch based on a random number.

Reply:
Let's leave as-is for now and if you're interested in experimenting we can entertain a PR modification later.

Reply:
Who knows, it is a matter of trying. In principle, it could be translated to branchless code (think of `i + rand(Bool)*(j-i)`).